### PR TITLE
Use the software selection cache in TUI

### PR DESCRIPTION
**Pull request by poncovka**
Use the DNF manager and the software selection cache to choose environments
and groups in the text user interface.

The cache remembers groups that were selected by default based on the current
environment, explicitly selected by the user or explicitly deselected. This is based
on the current implementation of the software selection in GUI.

The software selection in GUI will be refactorized in a follow-up pull request.

### CODE REVIEW

**Review by jkonecny12 — APPROVED** [review]
Looks like a great improvement! Thanks a lot for this. Just a few notes below.

File: pyanaconda/ui/lib/software.py [hunk 1]
```diff
@@ -33,3 +34,32 @@ def is_software_selection_complete(dnf_manager, selection, kickstarted=False):
...
     # The selected environment has to be valid.
     return dnf_manager.is_environment_valid(selection.environment)
+
+
+def get_software_selection_status(dnf_manager, selection, kickstarted=False):
+    """Get the software selection status.
+
+    :param dnf_manager: a DNF manager
+    :param selection: a packages selection data
+    :param kickstarted: is the selection configured by a kickstart file?
```
[the remark this answers is not in the document]
Comment (poncovka) [reply pyanaconda/ui/lib/software.py:46]: Fixed.

File: pyanaconda/ui/tui/spokes/software_selection.py [hunk 2]
```diff
@@ -148,24 +149,19 @@ def _get_available_addons(self, environment_id):
 
     @property
     def status(self):
-        """ Where we are in the process """
+        """The status of the spoke."""
         if self.errors:
             return _("Error checking software selection")
         if not self.ready:
-            return _("Processing...")
```
[the remark this answers is not in the document]
Comment (poncovka) [reply pyanaconda/ui/tui/spokes/software_selection.py:154]: Fixed.

**Comment by poncovka** [comment]
Updated.

**Review by jkonecny12 — APPROVED** [review]
Looks even better to me now, thanks!

**Pull request merged**